### PR TITLE
added implementations of MachineProducer and MachineService

### PR DESCRIPTION
**Pull request by BusyhinD**

### CODE REVIEW

File: src/main/java/core/mate/academy/service/MachineProducer.java [hunk 1]
```diff
@@ -1,11 +1,13 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 public interface MachineProducer {
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineProducer.java:6]: **interface is not parametrized**
image

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 2]
```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineService.java:20]:
```suggestion
    List<Machine> getAll(Class<? extends T> type);
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 3]
```diff
@@ -26,13 +27,13 @@ public interface MachineService {
      * @param machines - list of machines to be filled with value
      * @param value    - any object of machine sub class
      */
-    void fill(List<Object> machines, Object value);
+    void fill(List<? super Machine> machines, T value);
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineService.java:30]:
```suggestion
    void fill(List<? super T> machines, T value);
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 4]
```diff
@@ -26,13 +27,13 @@ public interface MachineService {
...
 
     /**
      * Call the method doWork() from each machine.
      * This method should be able to accept a list of bulldozers as well as list of trucks.
      *
      * @param machines - the list of machines
      */
-    void startWorking(List<Object> machines);
+    void startWorking(List<? extends Machine> machines);
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineService.java:38]:
```suggestion
    void startWorking(List<? extends T> machines);
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 5]
```diff
@@ -1,7 +1,47 @@
...
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> list) {
+        for (Machine machine : list) {
+            machine.doWork();
+        }
+    }
+
+    private MachineProducer compareTypeWithMachines(Class<? extends Machine> type) {
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:34]:
```suggestion
    private MachineProducer getProducerByType(Class<? extends Machine> type) {
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 6]
```diff
@@ -1,7 +1,47 @@
...
+        if (type == null) {
+            throw new RuntimeException("Required type is null");
+        } else if (type.equals(Bulldozer.class)) {
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:37]:
```suggestion
        if (Bulldozer.class.equals(type)) {
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**